File: Selection_sort.java
public class Selection_sort {
    public static void main(String[] args) {
        int[] arr={5,55,23,2,7,1,9};
        
        for(int i=0;i<arr.length-1;i++){
            int min=i;
            int j=i+1;
            for( ;j<arr.length;j++){
                if(arr[min]>arr[j])
                min=j;
            }
            arr[min] +=(arr[i]-(arr[i]=arr[min])); // Swaping the no..x+=(y-(y=x))
            //Another way by XOR  x= x^y^(y=x);

        }
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
    }
}

File: SelectionSort.cpp
#include<iostream>
using namespace std;
int main(){
    int arr[20]={77,5,9,12,3,7,1};
    for(int i=0;i<7;i++){
        int min=i;
        for(int j=i+1;j<7;j++){
            if(arr[j]<arr[min]){
                min=j;
            }
            
        }
        int temp=arr[i];
        arr[i]=arr[min];
        arr[min]=temp;
    }
    for(int i=0;i<7;i++){
        cout<<arr[i]<<" ";
    }
}